### PR TITLE
Replace justified `unwraps` in `shardtree` with `expect`s

### PR DESCRIPTION
**Pull request by str4d**
Part of #118.

### CODE REVIEW

File: shardtree/src/lib.rs [hunk 1]
```diff
@@ -406,13 +406,16 @@ impl<
 
     /// Adds a checkpoint at the rightmost leaf state of the tree.
     pub fn checkpoint(&mut self, checkpoint_id: C) -> Result<bool, ShardTreeError<S::Error>> {
+        /// Pre-condition: `root_addr` must be the address of `root`.
```
[the remark this answers is not in the document]
Comment (str4d) [reply shardtree/src/lib.rs:409]: > it is a programming error for these to be inconsistent.

I note that the public API `LocatedTree::from_parts` does not enforce that its arguments are consistent; we could potentially strengthen this by making that method fallible and checking that the actual encoded depth of the provided tree matches the provided root address (which would ensure that all parents within the provided tree correspond to a subtree node of the provided root address that can have children).

[the remark this answers is not in the document]
Comment (str4d) [reply shardtree/src/lib.rs:409]: I've updated this PR to add the consistency check to `LocatedTree::from_parts`.

File: shardtree/src/prunable.rs [hunk 2]
```diff
@@ -440,8 +446,10 @@ impl<H: Hashable + Clone + PartialEq> LocatedPrunableTree<H> {
     /// Returns either the witness for the leaf at the specified position, or an error that
     /// describes the causes of failure.
     pub fn witness(&self, position: Position, truncate_at: Position) -> Result<Vec<H>, QueryError> {
-        // traverse down to the desired leaf position, and then construct
-        // the authentication path on the way back up.
+        /// Traverse down to the desired leaf position, and then construct
+        /// the authentication path on the way back up.
+        //
```
Comment (daira) [inline shardtree/src/prunable.rs:451]: Non-blocking nit:
```suggestion
        ///
```

**Review by daira — APPROVED** [review]
utACK

File: shardtree/src/prunable.rs (hunk 2 above)
Comment (nuttycom) [reply shardtree/src/prunable.rs:451]: I didn't think that `///` comments were allowed on internal documentation? I suspect this whole comment will produce a linting or doc error.

**Review by nuttycom — APPROVED** [review]
utACK modulo doc/comment linting c05f2fac006b7336939c1854c774f3642f581214

File: shardtree/src/prunable.rs (hunk 2 above)
Comment (nuttycom) [reply shardtree/src/prunable.rs:451]: I verified that `cargo doc` accepts these without complaint.

**Pull request merged**